### PR TITLE
Update MasterServerPinger to modern approach

### PR DESCRIPTION
**Pull request by teinarss**

### CODE REVIEW

**Comment by pchote** [comment]
```
Checking runtime assemblies...
error: The following assemblies are referenced but not whitelisted:
   System.Net.Http.dll
```

The makefile whitelist and packaging scripts will need to include this!

File: packaging/functions.sh [hunk 1]
```diff
@@ -66,7 +66,7 @@ install_assemblies_mono() {
 	fi
 
 	# Managed Dependencies
-	for LIB in ICSharpCode.SharpZipLib.dll FuzzyLogicLibrary.dll Open.Nat.dll BeaconLib.dll DiscordRPC.dll Newtonsoft.Json.dll SDL2-CS.dll OpenAL-CS.Core.dll Eluant.dll; do
+	for LIB in ICSharpCode.SharpZipLib.dll FuzzyLogicLibrary.dll Open.Nat.dll BeaconLib.dll DiscordRPC.dll Newtonsoft.Json.dll SDL2-CS.dll OpenAL-CS.Core.dll Eluant.dll System.Net.Http.dll; do
```
Comment (pchote) [inline packaging/functions.sh]: Ugh... this doesn't actually work 😞  https://github.com/pchote/OpenRA/actions/runs/435506045

This leaves us with three options:
1. go back to one big PR that switches everything to net 5 in one go
2. update my mono support bundles for macOS and Linux to add this (and change the makefile back to WHITELISTED_CORE_ASSEMBLIES)
3. stay with `WebClient` until we've switched our release packages to Core (removing the dependency on 2)

1 isn't reasonable, and i'd rather not go to have to do 2 so close to finally getting rid of them, so can we please change this to just the `Task.Run` switch?

**Comment by teinarss** [comment]
Updated.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
The changelog for this should be part of a more generic "Switch Windows releases to .NET 5" entry.

**Review by abcdefg30 — APPROVED** [review]
Ok.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/1dbb6840497db19defd19d868bf1be5d168875fa...0ad6b9a69b8f5a9abcfe63d524d0affa5a5eac23)

**Pull request merged**